src/Control/Carrier/Fail/Either.hs

```diff
@@ -1,4 +1,9 @@
 {-# LANGUAGE FlexibleInstances, GeneralizedNewtypeDeriving, MultiParamTypeClasses, TypeOperators, UndecidableInstances #-}
+-- | A carrier for a 'Fail' effect, returning the result as an 'Either' 'String'. Failed computations
+-- will return a 'Left' containing the 'String' value passed to 'Control.Monad.Fail.fail'.
```

Comment:
We’ve imported the module qualified as `Fail`, so should be able to say just `Fail.fail` I think.